src/devices_models/devices/hydro_generation.jl

```diff
@@ -13,40 +13,26 @@ struct HydroCommitmentReservoirFlow <: AbstractHydroUnitCommitment end
 struct HydroCommitmentReservoirStorage <: AbstractHydroUnitCommitment end
 =#
 ########################### Hydro generation variables #################################
-function activepower_variables!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{H},
-) where {H <: PSY.HydroGen}
-    add_variable(
-        psi_container,
-        devices,
-        variable_name(ACTIVE_POWER, H),
-        false,
-        :nodal_balance_active;
-        lb_value = d -> PSY.get_activepowerlimits(d).min,
-        ub_value = d -> PSY.get_activepowerlimits(d).max,
-        init_value = d -> PSY.get_activepower(d),
+function make_active_power_add_variable_inputs(::Type{<:PSY.HydroGen}, ::PSIContainer)
+    return AddVariableInputs(;
+        variable_names = [ACTIVE_POWER],
+        binary = false,
+        expression_name = :nodal_balance_active,
+        initial_value_func = x -> PSY.get_activepower(x),
+        lb_value_func = x -> PSY.get_activepowerlimits(x).min,
+        ub_value_func = x -> PSY.get_activepowerlimits(x).max,
     )
-
-    return
 end
 
-function reactivepower_variables!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{H},
-) where {H <: PSY.HydroGen}
-    add_variable(
-        psi_container,
-        devices,
-        variable_name(REACTIVE_POWER, H),
-        false,
-        :nodal_balance_reactive;
-        ub_value = d -> PSY.get_reactivepowerlimits(d).max,
-        lb_value = d -> PSY.get_reactivepowerlimits(d).min,
-        init_value = d -> PSY.get_reactivepower(d),
+function make_reactive_power_add_variable_inputs(::Type{<:PSY.HydroGen}, ::PSIContainer)
```

Comment:
these are really long function names, I think we need something more descriptive and obvious.

Reply:
I'm open to suggestions.

Reply:
1. `mkvarinp_reactive_power`
 2. `mk_reactive_power_var_inp`
 3. `reactive_power_variable_inputs`

I like the ones that would have a consistent prefix for the different applications (1 & 2). We could break from our style guid a bit and abbreviate these processes, so we could make something like `avi` for `the `add_variable_inputs` or `mavi`, giving us:

 4. `make_reactive_power_avi`
 5. `mavi_reactive_power`

Reply:
Keep in mind that we already have
```
function make_active_power_constraints_inputs
function make_reactive_power_constraints_inputs
function make_nodal_expression_inputs
```
We can change those if necessary.  I do think that we should follow the same pattern everywhere.

Reply:
I suggest keeping just 

make_active_power_constraints
make_active_power_variables 

I Think a level 2 user doesn't need to know the whole inputs layer.

Reply:
Keep in mind that this function returns an instance of DeviceRangeConstraintInputs.  Here is the context of where it's used:

```
function activepower_constraints!(psi_container, devices, model)
    inputs = make_active_power_constraints_inputs(...)
    device_range_constraints!(psi_container, devices, model, feedforward, inputs)
```

Changing it to 
```
    inputs = make_active_power_constraints(...)
```
might confuse some readers.  Do you still have the same opinion?

Reply:
maybe just `make_reactive_power_variable_inputs`